dts/bindings/can/bosch\,m_can.yaml

```
@@ -0,0 +1,22 @@
+description: Bosch M_CAN CAN FD generic driver
+
+compatible: "bosch,m_can"
+
+include:
+  - name: bosch,m_can-base.yaml
+  - name: pinctrl-device.yaml
+
+properties:
+  reg:
+    required: true
+
+  clock-frequency:
+    type: int
+    description: |
+      Optional frequency to use if clocks are not defined
```

This is very nice, but is it really needed? I would prefer to avoid having clock frequencies listed for peripherals like this, and instead force use of the clock controller API.

```suggestion
      Optional frequency in Hertz to use if clocks are not defined
```

---

I can totally understand that, however sometimes the frequency can be configured in the bootloader (at least the one i am using right now), in which case I thought it would be better to program a static value. The ns16550 driver does something very similar. If the consensus is to remove it, I will get rid of it.